lightning/src/routing/network_graph.rs

```diff
@@ -1813,4 +1932,165 @@ mod tests {
 		network.write(&mut w).unwrap();
 		assert!(<NetworkGraph>::read(&mut ::std::io::Cursor::new(&w.0)).unwrap() == *network);
 	}
+
+	#[test]
+	fn calling_sync_routing_table() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+		let first_blocknum = 0;
+		let number_of_blocks = 0xffff_ffff;
+
+		// It should ignore if gossip_queries feature is not enabled
+		{
+			let init_msg = Init { features: InitFeatures::known().clear_gossip_queries() };
+			net_graph_msg_handler.sync_routing_table(&node_id_1, &init_msg);
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 0);
+		}
+
+		// It should send a query_channel_message with the correct information
+		{
+			let init_msg = Init { features: InitFeatures::known() };
+			net_graph_msg_handler.sync_routing_table(&node_id_1, &init_msg);
+			let events = net_graph_msg_handler.get_and_clear_pending_msg_events();
+			assert_eq!(events.len(), 1);
+			match &events[0] {
+				MessageSendEvent::SendChannelRangeQuery{ node_id, msg } => {
+					assert_eq!(node_id, &node_id_1);
+					assert_eq!(msg.chain_hash, chain_hash);
+					assert_eq!(msg.first_blocknum, first_blocknum);
+					assert_eq!(msg.number_of_blocks, number_of_blocks);
+				},
+				_ => panic!("Expected MessageSendEvent::SendChannelRangeQuery")
+			};
+		}
+	}
+
+	#[test]
+	fn handling_reply_channel_range() {
+		let (secp_ctx, net_graph_msg_handler) = create_net_graph_msg_handler();
+		let node_privkey_1 = &SecretKey::from_slice(&[42; 32]).unwrap();
+		let node_id_1 = PublicKey::from_secret_key(&secp_ctx, node_privkey_1);
+
+		let chain_hash = genesis_block(Network::Testnet).header.block_hash();
+
+		// Test receipt of a single reply that should enqueue an SCID query
+		// matching the SCIDs in the reply
+		{
+			// Handle a single successful reply that encompasses the queried channel range
+			let result = net_graph_msg_handler.handle_reply_channel_range(&node_id_1, ReplyChannelRange {
+				chain_hash,
+				full_information: true,
+				first_blocknum: 0,
+				number_of_blocks: 2000,
+				short_channel_ids: vec![
+					0x0003e0_000000_0000, // 992x0x0
+					0x0003e8_000000_0000, // 1000x0x0
+					0x0003e9_000000_0000, // 1001x0x0
+					0x0003f0_000000_0000, // 1008x0x0
+					0x00044c_000000_0000, // 1100x0x0
+					0x0006e0_000000_0000, // 1760x0x0
+				],
+			});
+			assert!(result.is_ok());
```

Comment:
Given `handle_reply_channel_range` doesn't verify anymore that scid replies are bounded in queried channel range is this test check still useful ?

Reply:
Yep, the test is still valid, but the comment on line 1983 is outdated, will clean that up.